### ISSUE TITLE
Merge Without Commit

### ISSUE THREAD

**Issue by surmann** [jesseduffield/lazygit#650]
Sometimes I use the option "--no-commit" when merging branches. Is it possible to with lazygit? I just didn't find this option.
Thanks!

**Comment by jesseduffield** [comment]
This functionality is not yet available, though I suspect these kinds of issues will continue coming up i.e. 'I want to do this thing in lazygit but passing an additional flag. I wonder how best to display that in the UI. One option is to, after the user has selected to merge a branch, show another menu containing a list of optional flags you may want to pass. This unfortunately means an additional keypress which I would rather avoid, but I can't think of too many other options. A  couple I can think of:
1) Add another option saying 'merge with additional flags' in the original menu. This might bloat our menus with duplicate options if we end up needing to add it in many places
2) have some way of indicating that the current menu item has an 'advanced' mode where if you use a special keybinding other than enter, then you'll get the multi-choice additional menu popup.

What are your thoughts?

**Comment by corytheboyd-smartsheet** [comment]
> This functionality is not yet available, though I suspect these kinds of issues will continue coming up i.e. 'I want to do this thing in lazygit but passing an additional flag. I wonder how best to display that in the UI. One option is to, after the user has selected to merge a branch, show another menu containing a list of optional flags you may want to pass. This unfortunately means an additional keypress which I would rather avoid, but I can't think of too many other options. A couple I can think of:
> 
> 1. Add another option saying 'merge with additional flags' in the original menu. This might bloat our menus with duplicate options if we end up needing to add it in many places
> 2. have some way of indicating that the current menu item has an 'advanced' mode where if you use a special keybinding other than enter, then you'll get the multi-choice additional menu popup.
> 
> What are your thoughts?

Yeah all those concerns make sense, and I really enjoy the opinions that lazygit already has (as opposed to exposing every permutation of every flag as distinct options).

Would it be absolutely insane to introduce a new modifier key to engage such an advanced mode on commands that support it? For example, `cmd + shift + <KEY>`. Commands that support this would need some indicator of course. Agree that keeping the default path clear of extra key presses is the right call though, and it's really not the end of the world to go orchestrate my own flows when they don't exactly line up with the purpose of lazygit.
